packages/compiler/src/server/serverlib.ts

```diff
@@ -339,6 +343,113 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  async function updateImportsOnFileChange(changes: FileEvent[]): Promise<void> {
+    const tspChanges = changes.filter((change) => change.uri.endsWith(".tsp"));
+    const filePathChangedMap = new Map<string, string>();
+
+    if (tspChanges.length === 0 || tspChanges.length % 2 !== 0) {
+      return;
+    }
+
+    // type 1: created, type 3: deleted
+    // Modify file name or move the file to a new location,
+    // the original file is marked as delete, and the new file is marked as Create
+    if (tspChanges.length === 2) {
+      // Only one file is renamed or moved
+      const oldFilePath = tspChanges.find((c) => c.type === 3);
+      const newFilePath = tspChanges.find((c) => c.type === 1);
+      if (oldFilePath && newFilePath) {
+        filePathChangedMap.set(
+          await fileService.getPath({ uri: oldFilePath.uri }),
+          await fileService.getPath({ uri: newFilePath.uri }),
+        );
+      }
+    } else {
+      for (const change of tspChanges) {
+        if (change.type === 1) {
```

Comment:
don't use the magic numbers replace with `FileChangeType.Createad` or `FileChangeType.Deleted`